src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVTextMouseActionData.cs

```diff
@@ -0,0 +1,544 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using RegressionGames.StateRecorder.BotSegments.JsonConverters;
+using RegressionGames.StateRecorder.BotSegments.Models.CVService;
+using RegressionGames.StateRecorder.JsonConverters;
+using RegressionGames.StateRecorder.Models;
+using UnityEngine;
+
+namespace RegressionGames.StateRecorder.BotSegments.Models
+{
+    /**
+     * <summary>Data for clicking on or moving the mouse to a CV Text location in the scene.
+     * This processes the results by finding the smallest bounding rect of the matching texts possible withinRect if specified and clicking on the center of that rect.</summary>
+     */
+    [Serializable]
+    public class CVTextMouseActionData : IBotActionData
+    {
+        // api version for this object, update if object format changes
+        public int apiVersion = SdkApiVersion.VERSION_14;
+
+        [NonSerialized]
+        public static readonly BotActionType Type = BotActionType.Mouse_CVText;
+
+        public string text;
+        public TextMatchingRule textMatchingRule = TextMatchingRule.Matches;
+        public TextCaseRule textCaseRule = TextCaseRule.Matches;
+
+        /**
+         * optionally limit the rect area where the TextData can be detected
+         */
+        [CanBeNull]
+        public CVWithinRect withinRect;
+
+        /**
+         * the list of actions to perform at this location
+         * Note: This does NOT re-evaluate the position of the CVText on each action.
+         * This is modeled as a list to allow you to click and/or un-click buttons on this Text location without re-evaluating the CV text match in between if you don't want to.
```

Comment:
Oh awesome, I was curious about this. I wonder if we could also have some caching for scenarios where the criteria and action are both using CV on the same image... (obviously a discussion for another time)